packages/cardano-services/test/StakePoolSeach/StakePoolSearchHttpService.test.ts

```diff
@@ -468,6 +483,108 @@ describe('StakePoolSearchHttpService', () => {
           });
         });
       });
+
+      describe('stake pools sort', () => {
+        const filterArgs: StakePoolQueryOptions = {
+          filters: {
+            identifier: {
+              _condition: 'or',
+              values: [
+                { ticker: 'TEST' },
+                { name: STAKE_POOL_NAME },
+                { id: Cardano.PoolId('pool1jcwn98a6rqr7a7yakanm5sz6asx9gfjsr343mus0tsye23wmg70') }
+              ]
+            }
+          }
+        };
+
+        const sortByNameThenByPoolId = function (poolA: Cardano.StakePool, poolB: Cardano.StakePool) {
+          if ((poolA.metadata?.name || poolA.id) > (poolB.metadata?.name || poolB.id)) {
+            return 1;
+          } else if ((poolA.metadata?.name || poolA.id) < (poolB.metadata?.name || poolB.id)) {
+            return -1;
+          }
+          return 0;
+        };
+
+        it('sort by name desc order', async () => {
+          const response = await doServerRequest(setSortCondition({}, 'desc', 'name'));
+          expect(response).toMatchSnapshot();
+        });
+
+        it('sort by name asc order', async () => {
+          const response = await doServerRequest(setSortCondition({}, 'asc', 'name'));
+          expect(response).toMatchSnapshot();
+        });
+
+        it('if sort not provided defaults order by name and then by pool id asc', async () => {
+          const response = await doServerRequest({});
+
+          const resultSortedCopy = [...response.pageResults].sort(sortByNameThenByPoolId);
+
+          expect(response.pageResults).toEqual(resultSortedCopy);
+          expect(response).toMatchSnapshot();
+        });
+
+        it('if stake pool name is not presented, order should be then by pool id', async () => {
+          const fistNoMetadataPoolId = Cardano.PoolId('pool126zlx7728y7xs08s8epg9qp393kyafy9rzr89g4qkvv4cv93zem');
+          const secondNoMetadataPoolId = Cardano.PoolId('pool1y25deq9kldy9y9gfvrpw8zt05zsrfx84zjhugaxrx9ftvwdpua2');
```

Comment:
```suggestion
          const secondNoMetadataPoolId = Cardano.PoolId('pool1y25deq9kldy9y9gfvrpw8zt05zsrfx84zjhugaxrx9ftvwdpua2');
          const firstNamedPoolId = Cardano.PoolId('...');
          const secondNamedPoolId = Cardano.PoolId('...');
```

Reply:
Let's cover this scenario where there's a mix of named pools and unnamed pools, as that's the real-world scenario